### DISCUSSION TITLE
issue: infinite render loop is happen when use useWatch in split useFieldArray components

### DISCUSSION THREAD

**Discussion by tomoyukikino** [react-hook-form/react-hook-form#9494] [Q&A] [answered]
### Version Number

7.40.0

### Codesandbox/Expo snack

https://codesandbox.io/s/usefieldarray-react-bootstrap-hyc40z?file=/src/App.tsx

### Steps to reproduce

1. [Go to 'App.tsx' in codesandbox](https://codesandbox.io/s/usefieldarray-react-bootstrap-hyc40z?file=/src/App.tsx)
2. Scroll down to line 108
3. open Console
4. comment `<CartsFieldArrayForm />`
5. uncomment  `<CartsFromCard />` component
6. infinite render loop is happen!!

https://user-images.githubusercontent.com/45547251/205199175-61290d7d-2971-4112-a47e-94ae839933df.mp4

### Expected behaviour

Even if in split code in same function, infinite render loop better not happen.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
Thanks for the detailed issue report. 

The following inline component is getting unmounted and remounted on each render, please move into the render function instead.

![Screenshot 2022-12-02 at 1 22 25 pm](https://user-images.githubusercontent.com/10513364/205200322-9f9cd18f-2053-4e9f-a506-c4895feed85f.png)

**Reply by tomoyukikino** [reply]
I can solve split the component to another file.
I never knew the behavior about inline components.
Surely, There is not issue.
I notice I need to more study about react.

**Reply by bluebill1049** [reply]
That's all good, we've all been there.

**Reply by tomoyukikino** [reply]
you are so kind!
